core/src/validator.rs

```diff
@@ -1306,7 +1306,7 @@ impl Validator {
             config.wait_to_vote_slot,
             accounts_background_request_sender,
             config.runtime_config.log_messages_bytes_limit,
-            &connection_cache,
+            json_rpc_service.is_some().then_some(&connection_cache), // for the cache warmer only used for STS for RPC service
```

Comment:
It looks like we can avoid creating connection_cache entirely if json_rpc_service.is_none()?

Reply:
we can't, the forwarder still uses the cache